apps/docs/pages/docs/api-reference/fields/richtext.mdx

```diff
@@ -0,0 +1,733 @@
+        variant={editorState.isCodeBlock ? "secondary" : "primary"}
+        title="Code block"
+      >
+        <SquareCode size={20} />
+      </IconButton>
+    ),
+  },
+
+  Blockquote: {
+    render: (editor: Editor, editorState: EditorState) => (
+      <IconButton
+        onClick={(e) => {
+          e.stopPropagation();
+          editor.chain().focus().toggleBlockquote().run();
+        }}
+        disabled={!editorState.canBlockquote}
+        variant={editorState.isBlockquote ? "secondary" : "primary"}
+        title="Blockquote"
+      >
+        <Quote size={20} />
+      </IconButton>
+    ),
+  },
+
+  HorizontalRule: {
+    render: (editor: Editor, editorState: EditorState) => (
+      <IconButton
+        onClick={(e) => {
+          e.stopPropagation();
+          editor.chain().focus().setHorizontalRule().run();
+        }}
+        disabled={!editorState.canHorizontalRule}
+        variant="primary"
+        title="Horizontal rule"
+      >
+        <Minus size={20} />
+      </IconButton>
+    ),
+  },
+}
+```
+
+### `textSelectOptions`
+
+```tsx
+["p", "h2", "h3", "h4", "h5", "h6"];
+```
+
+### `extensions`
+
+```tsx
+["StarterKit", "TextAlign"];
+```
+
+### `selector`
+
+```tsx
+(ctx: EditorStateSnapshot) => {
+  const editor = ctx.editor;
+  if (!editor) return {};
+
+  return {
+    isAlignLeft: editor.isActive({ textAlign: "left" }),
+    canAlignLeft: editor.can().chain().setTextAlign("left").run(),
+
+    isAlignCenter: editor.isActive({ textAlign: "center" }),
+    canAlignCenter: editor.can().chain().setTextAlign("center").run(),
+
+    isAlignRight: editor.isActive({ textAlign: "right" }),
+    canAlignRight: editor.can().chain().setTextAlign("right").run(),
+
+    isAlignJustify: editor.isActive({ textAlign: "justify" }),
+    canAlignJustify: editor.can().chain().setTextAlign("justify").run(),
+
+    isBold: editor.isActive("bold"),
+    canBold: editor.can().chain().toggleBold().run(),
+
+    isItalic: editor.isActive("italic"),
+    canItalic: editor.can().chain().toggleItalic().run(),
+
+    isUnderline: editor.isActive("underline"),
+    canUnderline: editor.can().chain().toggleUnderline().run(),
+
+    isStrike: editor.isActive("strike"),
+    canStrike: editor.can().chain().toggleStrike().run(),
+
+    isInlineCode: editor.isActive("code"),
+    canInlineCode: editor.can().chain().toggleCode().run(),
+
+    isBulletList: editor.isActive("bulletList"),
+    canBulletList: editor.can().chain().toggleBulletList().run(),
+
+    isOrderedList: editor.isActive("orderedList"),
+    canOrderedList: editor.can().chain().toggleOrderedList().run(),
+
+    isCodeBlock: editor.isActive("codeBlock"),
+    canCodeBlock: editor.can().chain().toggleCodeBlock().run(),
+
+    isBlockquote: editor.isActive("blockquote"),
+    canBlockquote: editor.can().chain().toggleBlockquote().run(),
+
+    canHorizontalRule: editor.can().chain().setHorizontalRule().run(),
+  };
+};
+```
+
+## Extending the Editor
```

Comment:
These usage guidelines should be in "Integrating/Extending Puck", not in the API reference.